api/main.py

```diff
@@ -101,6 +101,33 @@ async def get_node(node_id: str):
     return await db.find_by_id(Node, node_id)
 
 
+@app.get('/nodes')
+async def nodes():
+    return {'nodes': await db.find_all(Node)}
+
+
+@app.get('/node/{attr_name}/{attr_value}', response_model=Node)
+async def get_node(attr_name: str, attr_value: str):
+    return await db.find_by_attribute(Node, attr_name, attr_value)
+
+
+@app.get('/node/revision/{attr_name}/{attr_value}', response_model=Node)
+async def get_node(attr_name: str, attr_value: str):
+    return await db.find_by_attribute(Node, "revision", attr_name, attr_value)
```

Comment:
This is a bit too specific to `Node.revision`, we can use the dotted syntax instead e.g. `revision.attr=value`.  In fact we may be able to pass it as-is when doing a Mongo DB query as it's supported as well.  For example:
```
> db.node.findOne({'revision.describe': 'v5.16-rc5-224-g9eaa88c7036e'})
{
	"_id" : ObjectId("61bdb69baddf9068ffcb7b37"),
	"kind" : "node",
	"name" : "checkout",
	"revision" : {
		"tree" : "mainline",
		"url" : "https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git",
		"branch" : "master",
		"commit" : "9eaa88c7036eda3f6c215f87ca693594cf90559b",
		"describe" : "v5.16-rc5-224-g9eaa88c7036e"
	},
	"parent" : null,
	"status" : true
}
```